### PR TITLE
Various documentation updates

### PR DESCRIPTION
**Pull request by whzup**
Created an RST file for the Inverse Kinematics tutorial and fixed some typos in the text.
Added citation in the Pyramid class.
Changed the subject line of doc strings to a commanding style.
Simplified the doc strings using the http://www.hemingwayapp.com/.

See also: #154

### CODE REVIEW

**Comment by whzup** [comment]
The LaTex script does not work properly as in the other RST-files, too. It may have something to do with [this](https://stackoverflow.com/questions/3610551/math-in-restructuredtext-with-latex).

File: pyswarms/backend/topology/pyramid.py [hunk 1]
```diff
@@ -28,8 +28,8 @@ def __init__(self):
     def compute_gbest(self, swarm):
         """Update the global best using a pyramid neighborhood approach
 
-        This uses the Delaunay method from :code:`scipy` to triangulate N-dimensional space
-        with simplices consisting of swarm particles [SIS2008]
+        This topology uses the Delaunay method from :code:`scipy`. This method is used to triangulate
+        N-dimensional space into simplices. The vertecies of the simplicies consist of swarm particles. [SIS2008]
```
Comment (ljvmiranda921) [inline pyswarms/backend/topology/pyramid.py]: vertecies to vertices :+1:

**Comment by ljvmiranda921** [comment]
For reference: 
https://stackoverflow.com/questions/20928727/adding-commits-to-another-persons-pull-request-on-github

**Comment by ljvmiranda921** [comment]
And now we're down to this haha! Let me try this next week, might need a lot of work. The pyup bot also made a lot of PRs and some of them are failing (I wonder why, afaik it's because of a broken dependency in pyyaml), I'll merge all of them into development then to master then we'll draft the release notes by august :+1:

File: pyswarms/single/__init__.py [hunk 2]
```diff
@@ -2,6 +2,16 @@
...
+
+.. note::
+    PSO algorithms scale with the search space. This means that, by
+    using larger boundaries, the final results are getting larger
+    as well.
+
+.. note::
+    Please keep in mind that Python has a biggest float number.
+    So using large boundaries in combination with exponentiation or
+    multiplication can lead to an :code:`OverflowError`.
```
Comment (ljvmiranda921) [inline pyswarms/single/__init__.py]: Yes, this one is good. :+1:

**Comment by ljvmiranda921** [comment]
I tried resolving some merge conflicts and it passed our CI. Would you mind double-checking on your end, @whzup ? If this is all good, we can merge now and make a PR for the development branch (and v.0.4.0 preparation).

**Pull request merged**